### ISSUE TITLE
Synchronously return a stream?

### ISSUE THREAD

**Issue by neumino** [neumino/rethinkdbdash#61]
Comment on reddit from SubStack:
http://www.reddit.com/r/node/comments/2uj3fk/rethinkdbdash_116_nodejs_streams_implicit_run_and/co9crh9

> This part seems slightly strange:
````
r.table("data").then(function(stream) {
    // Pipe the data to the file through stringifier
    stream.pipe(stringifier).pipe(file);
}).error(function(error) {
    // Handle error
});`
```
a promise provides a stream? streams already have their own error propagation mechanisms, so this seems unnecessarily indirect and duplicates error handling logic. I'm not big on promises in the first place, but their featureset seems exclusive to streams.


So we could have something like

```
r.table('data').toStream().pipe(stringifier).pipe(file);
```

That seems like a cleaner syntax. There might be a few edge cases, but it seems like a reasonable idea at least.

**Comment by neumino** [comment]
(accidentally hit post a bit too early. I just edited the original issue)

**Comment by neumino** [comment]
We could also just have `pipe` as a shortcut for `toStream().pipe`.

**Comment by neumino** [comment]
Now that I think about it, the first time the event readable is produced is when the promise is resolved.

Does it still make sense in some cases for run to return a promise that will be resolve with a stream?

**Comment by neumino** [comment]
`pipe` is probably a bad idea as people probably want to add a listener for `error`, but `toStream` still sounds like a good idea.

**Comment by marshall007** [comment]
> Does it still make sense in some cases for run to return a promise that will be resolve with a stream?

I don't believe so, I've never seen an API exposed this way before. `query.toStream().pipe(...)` is the way to go, IMO.

**Comment by neumino** [comment]
Shipped in 1.16.4.
`run({stream: true})` is deprecated, and will eventually be removed except if someone has a good reason not too. But basically the event "readable" is fired when the query is returned, so there's little reason to require error handler in multiple places.
